regression/cbmc/r_w_ok1/main.c

```diff
@@ -0,0 +1,26 @@
+#include <assert.h>
+#include <stdlib.h>
+
+int main()
+{
+  int *p = NULL;
+
+  assert(!__CPROVER_r_ok(p, sizeof(int)));
+  assert(!__CPROVER_w_ok(p, sizeof(int)));
+
+  p = malloc(sizeof(int));
+
+  assert(__CPROVER_r_ok(p, 1));
+  assert(__CPROVER_w_ok(p, 1));
+  assert(__CPROVER_r_ok(p, sizeof(int)));
+  assert(__CPROVER_w_ok(p, sizeof(int)));
+
+  size_t n;
+  char *arbitrary_size = malloc(n);
+
+  assert(__CPROVER_r_ok(arbitrary_size, n));
+  assert(__CPROVER_w_ok(arbitrary_size, n));
+
+  assert(__CPROVER_r_ok(arbitrary_size, n + 1));
```

Comment:
Why does this succeed?

Reply:
Without `--pointer-check` any use of `__CPROVER_{r,w}_ok` is now a no-op (and is now also documented as such), but with `--pointer-check` this fails as expected (there are two regression tests using the same `main.c` file included).

Reply:
Is this what we want? I would find it more intuitive if these were independent.

Reply:
It does make the implementation a lot easier, and also removes a need for documentation, so that seems much better. The only cost is that we will unconditionally need to run the local alias analysis.

Reply:
Ok, this will require #4471 to be merged first. And another bug left to be debugged.

Reply:
Hmm, actually there is more to it than just #4471, we also seem to be messing up `ID_this` elsewhere.

Reply:
If the cost of the local alias analysis is a concern, there could be a quick scan for the predicate in the function; but so far, that analysis has not been cause for concern.

Reply:
Agreed, I just need to fix bugs in the C++ and Java front-ends first as the local alias analysis needs to succeed in doing symbol-table lookups...